src/sidebar/test/oauth-auth-test.js

```diff
@@ -58,16 +66,35 @@ describe('oauth auth', function () {
       });
     });
 
-    it('should raise if an access token request fails', function () {
-      fakeHttp.post.returns(Promise.resolve({status: 500}));
-      return auth.tokenGetter().then(
-        function onResolved () {
-          assert(false, 'The Promise should have been rejected');
-        },
-        function onRejected (error) {
+    context('when the access token request fails', function() {
+      beforeEach('make access token requests fail', function () {
```

Comment:
Yeah, it's used in tracebacks, but I think also serves to make the code more readable (maybe). You can also pass it a named function, does the same thing. This also implies that if you want to do multiple things maybe it should be multiple separate `beforeEach`s.